Sources/Overload/OvEditor/src/OvEditor/Core/GizmoBehaviour.cpp

```diff
@@ -108,22 +110,38 @@ OvMaths::FVector2 OvEditor::Core::GizmoBehaviour::GetScreenDirection(const OvMat
 	return OvMaths::FVector2::Normalize(result);
 }
 
-void OvEditor::Core::GizmoBehaviour::ApplyTranslation(const OvMaths::FMatrix4& p_viewMatrix, const OvMaths::FMatrix4& p_projectionMatrix, const OvMaths::FVector2& p_viewSize) const
+void OvEditor::Core::GizmoBehaviour::ApplyTranslation(const OvMaths::FMatrix4& p_viewMatrix, const OvMaths::FMatrix4& p_projectionMatrix, const OvMaths::FVector2& p_viewSize)
 {
-	auto unitsPerPixel = 0.001f * m_distanceToActor;
-	auto originPosition = m_originalTransform.GetWorldPosition();
+	auto ray = GetMouseRay(m_currentMouse, p_viewMatrix, p_projectionMatrix, p_viewSize);
 
-	auto screenDirection = GetScreenDirection(p_viewMatrix, p_projectionMatrix, p_viewSize);
+	const OvMaths::FVector3 planeTangent = OvMaths::FVector3::Cross(GetRealDirection(true), m_target->transform.GetWorldPosition() - m_cameraPosition);
+	const OvMaths::FVector3 planeNormal = OvMaths::FVector3::Cross(GetRealDirection(true), planeTangent);
 
-	auto totalDisplacement = m_currentMouse - m_originMouse;
-	auto translationCoefficient = OvMaths::FVector2::Dot(totalDisplacement, screenDirection) * unitsPerPixel;
+	OvMaths::FVector3 direction = GetRealDirection(true);
 
-	if (IsSnappedBehaviourEnabled())
+	const OvMaths::FVector3 planePoint = m_originalTransform.GetWorldPosition();
+
+	const float denom = OvMaths::FVector3::Dot(ray.second, planeNormal);
+
+	if (std::abs(denom) == 0)
```

Comment:
Comparing a float to an int here, maybe we could check if the `abs(denom)` is `< some_small_float`:

```suggestion
	if (std::abs(denom) <= 0.001f)
```

Reply:
Or maybe it's fine, and we could compare it with `0.0f`